ydb/core/tx/scheme_board/monitoring.cpp

```diff
@@ -1294,18 +1383,137 @@ class TMonitoring: public TActorBootstrapped<TMonitoring> {
+                ));
+            }
+
+            if (params.Has("backupProgress")) {
+                return (void)Send(ev->Sender, new NMon::TEvHttpInfoRes(
+                    TStringBuilder() << NMonitoring::HTTPOKJSON << BackupProgress.ToJson(),
+                    0, EContentType::Custom
+                ));
+            }
+
+            return (void)Send(ev->Sender, new NMon::TEvHttpInfoRes(RenderBackup(BackupProgress)));
+
+        case ERequestType::Restore:
+            if (params.Has("startRestore")) {
+                if (RestoreProgress.IsRunning()) {
+                    return (void)Send(ev->Sender, new NMon::TEvHttpInfoRes(
+                        RenderRestore(RestoreProgress, false, "Restore is already running")
+                    ));
+                }
+
+                TString filePath = params.Get("restorePath");
+                if (filePath.empty()) {
+                    return (void)Send(ev->Sender, new NMon::TEvHttpInfoRes(
+                        RenderRestore(RestoreProgress, false, "Restore file path is required")
+                    ));
+                }
+
+                ui64 schemeShardId = 0;
+                if (!TryFromString(params.Get("schemeShardId"), schemeShardId)) {
+                    return (void)Send(ev->Sender, new NMon::TEvHttpInfoRes(
+                        RenderRestore(RestoreProgress, false, "Invalid Scheme Shard ID")
+                    ));
+                }
+
+                ui64 generation = 1;
+                if (params.Has("generation")) {
+                    TryFromString(params.Get("generation"), generation);
+                }
+
+                RestoreProgress = TRestoreProgress();
+                RestoreProgress.Status = TRestoreProgress::EStatus::Starting;
+
+                SBB_LOG_I("Starting restore from " << filePath
+                    << " for SchemeShard ID: " << schemeShardId
+                    << " of generation: " << generation
+                );
+
+                Register(CreateSchemeBoardRestorer(filePath, schemeShardId, generation, SelfId()));
+
+                return (void)Send(ev->Sender, new NMon::TEvHttpInfoRes(
+                    RenderRestore(RestoreProgress, true)
+                ));
+            }
+
+            if (params.Has("restoreProgress")) {
+                return (void)Send(ev->Sender, new NMon::TEvHttpInfoRes(
+                    TStringBuilder() << NMonitoring::HTTPOKJSON << RestoreProgress.ToJson(),
+                    0, EContentType::Custom
+                ));
+            }
+
+            return (void)Send(ev->Sender, new NMon::TEvHttpInfoRes(RenderRestore(RestoreProgress)));
+
         case ERequestType::Unknown:
             break;
         }
 
         Send(ev->Sender, new NMon::TEvHttpInfoRes(NMonitoring::HTTPNOTFOUND, 0, EContentType::Custom));
     }
 
+    template <typename TProgress, typename TEvent>
+    void Handle(TProgress& progress, TEvent::TPtr& ev) {
```

Comment:
```suggestion
    template <typename TProgress, typename TEventPtr>
    void Handle(TProgress& progress, TEventPtr& ev) {
```

Тогда `TEventPtr` будет выводится автоматически.